[SYCL] linking sycl gives missing __cpu_model

On Fedora 28 + gcc 8.2.1

CMakeFiles/sycl.dir/source/detail/device_info.cpp.o: In function `cl::sycl::detail::get_native_vector_width(unsigned long)':
device_info.cpp:(.text._ZN2cl4sycl6detail23get_native_vector_widthEm+0x3): undefined reference to `__cpu_model'

I can fix this by manually adding -lgcc to the end of the c++ linker line, but that probably means something has gone wrong elsewhere, just mentioning here in case anyone else hits it.

--- a/sycl/CMakeLists.txt
+++ b/sycl/CMakeLists.txt
@@ -182,11 +182,10 @@ add_custom_target( sycl-toolchain
 
 set_target_properties("${SYCLLibrary}" PROPERTIES LINKER_LANGUAGE CXX)
 
-# Workaround for bug in GCC version 5.
+# Workaround for bug in GCC version 5 and higher.
 # More information https://bugs.launchpad.net/ubuntu/+source/gcc-5/+bug/1568899
 if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
-    CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.0 AND
-    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6.0)
+    CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.0)
   target_link_libraries("${SYCLLibrary}" gcc_s gcc)
 endif()
 

